scripts/ci_setup.py

```diff
@@ -0,0 +1,202 @@
+    This function writes a semaphore file when tools have been successfully
+    installed in order to avoid reinstalling them on every rerun.
+    '''
+    getLogger().info('Installing tools.')
+    channels = [
+        micro_benchmarks.FrameworkAction.get_channel(
+            target_framework_moniker)
+        for target_framework_moniker in target_framework_monikers
+    ]
+    dotnet.install(
+        architecture=architecture,
+        channels=channels,
+        versions=dotnet_versions,
+        verbose=verbose,
+    )
+
+def add_arguments(parser: ArgumentParser) -> ArgumentParser:
+    '''Adds new arguments to the specified ArgumentParser object.'''
+
+    if not isinstance(parser, ArgumentParser):
+        raise TypeError('Invalid parser.')
+
+    # Download DotNet Cli
+    dotnet.add_arguments(parser)
+    micro_benchmarks.add_arguments(parser)
+
+    parser.add_argument(
+        '--branch',
+        dest='branch',
+        required=False,
+        type=str,
+        help='Product branch.'
+    )
+    parser.add_argument(
+        '--commit-sha',
+        dest='commit_sha',
+        required=False,
+        type=str,
+        help='Product commit sha.'
+    )
+    parser.add_argument(
+        '--repository',
+        dest='repository',
+        required=False,
+        type=str,
+        help='Product repository.'
+    )
+    parser.add_argument(
+        '--queue',
+        dest='queue',
+        default='testQueue',
+        required=False,
+        type=str,
+        help='Test queue'
+    )
+    parser.add_argument(
+        '--build-number',
+        dest='build_number',
+        default='1234.1',
+        required=False,
+        type=str,
+        help='Build number'
+    )
+    
+    parser.add_argument(
+        '--locale',
+        dest='locale',
+        default='en-US',
+        required=False,
+        type=str,
+        help='Locale'
+    )
+    parser.add_argument(
+        '--perf-hash',
+        dest='perf_hash',
+        default='testSha',
+        required=False,
+        type=str,
+        help='Test queue'
+    )
+
+    parser.add_argument(
+        '--output-file',
+        dest='output_file',
+        required=False,
+        default=os.path.join(get_tools_directory(),'machine-setup' + global_extension),
+        type=str,
+        help='Filename to write the setup script to'
+    )
+
+    # Generic arguments.
+    parser.add_argument(
+        '-q', '--quiet',
+        required=False,
+        default=False,
+        action='store_true',
+        help='Turns off verbosity.',
+    )
+
+    parser.add_argument(
+        '--build-configs',
+        dest="build_configs",
+        required=False,
+        nargs='+',
+        default=[],
+        help='Configurations used in the build in key=value format'
+    )
+
+    return parser
+
+
+def __process_arguments(args: list):
+    parser = ArgumentParser(
+        description='Tool to run .NET micro benchmarks',
```

Comment:
is this copied from somewhere? Doesn't seem like the right description.

Reply:
Yup.